### PR TITLE
Addon-docs: Fix object array in Props

### PR DESCRIPTION
**Pull request by shilman**
Issue: #10499 #10498 

## What I did

- [x] Fix the array type. Before the "value" of the array was an array of types. Now it's a single type, which I think is more correct.
- [x] Change the control inference to show a basic JSON object editor when an array element has a complex type (e.g. object / unknown). Array control should only be used for arrays of scalars.

NOTE: The UX on this leaves a lot to be desired. This is a step up from the current state of things, though. cc @domyen 

## How to test

official-storybook:
- I added a story to test the "array of scalars" case
- There is an "array of react element" case already in the same file

### CODE REVIEW

File: addons/docs/src/frameworks/common/inferArgTypes.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ const inferType = (value?: any): SBType => {
   if (Array.isArray(value)) {
     const childType: SBType =
       value.length > 0 ? inferType(value[0]) : { name: 'other', value: 'unknown' };
```
Comment (tmeasday) [inline addons/docs/src/frameworks/common/inferArgTypes.ts:18]: So we don’t check every value in the array has the same type?

File: addons/docs/src/frameworks/common/inferControls.ts [hunk 2]
```diff
@@ -4,21 +4,27 @@ import { Control } from '@storybook/components';
...
     // console.log('no sbtype', { argType });
     return null;
   }
-  switch (argType.type.name) {
-    case 'array':
+  switch (type.name) {
+    case 'array': {
+      const { value } = type;
+      if (value?.name && ['object', 'other'].includes(value.name)) {
+        return { type: 'object' };
```
Comment (tmeasday) [inline addons/docs/src/frameworks/common/inferControls.ts:16]: I think we discussed doing some type checks when you edit the control to make sure that the object you entered is not only valid js but also an array. What are you thinking about that @shilman ?

**Review by tmeasday — COMMENTED** [review]
Couple of questions looks good otherwise

**Review by tmeasday — COMMENTED** [review]
Couple of questions looks good otherwise

File: addons/docs/src/frameworks/common/inferArgTypes.ts (hunk 1 above)
Comment (shilman) [reply addons/docs/src/frameworks/common/inferArgTypes.ts:18]: Not currently. WDYT is a good behavior in the case that they don't match? union type? error? any?

File: addons/docs/src/frameworks/common/inferControls.ts (hunk 2 above)
Comment (shilman) [reply addons/docs/src/frameworks/common/inferControls.ts]: Good point. Updated with "must be array" validation, but not sure whether using a functional validator will bite us later when we try to put this in the manager

**Pull request merged**